readme article (#3)
RoyNoymanW/order-custom-action: README.md

## About Wix app templates

[Wix apps](https://dev.wix.com/docs/build-apps) enhance the functionality of Wix sites by adding new features such as custom pages, dashboard components, third-party integrations, or site analytics. Starting with an app template fast-tracks the development process, providing a working foundational app that developers can modify and build upon. This approach saves valuable time, allowing for a quick transition from concept to a fully functional app.

Learn more about [Wix app templates](https://dev.wix.com/docs/build-apps/get-started/templates/get-started-from-an-app-template) and explore our growing [template collection](https://dev.wix.com/apps-templates).

## Template features

This Wix app template incorporates the following features:

- **Wix CLI:** Get a comprehensive developer experience with minimal setup and host your app on Wix with one command. Learn more about the [Wix CLI for apps](https://dev.wix.com/docs/build-apps/developer-tools/cli/get-started/about-the-wix-cli-for-apps).
- **Wix Design System:** Utilize Wix's reusable React components for a cohesive user experience consistent with Wix's design standards.

> - **Wix eCommerce Orders API**: Access and manage Wix eCommerce Orders data on a Wix site (this data is for orders made through the Wix Stores app). Learn more about the [Wix eCommerce API](https://dev.wix.com/docs/sdk/backend-modules/ecom/introduction).

Reviewer: Do we need the parenthetical comment here? eCom Orders can (in principle) also be used with orders created with Bookings, Restaurants, etc.

If so, maybe be more specific and say that the eCom Orders API is for use with all Wix eCommerce flows, but in this case it's for use with products sold via Stores.

- **Backend API**: Define your own HTTP functions that can be called from your frontend code. Learn more about [Backend APIs](https://dev.wix.com/docs/build-apps/develop-your-app/frameworks/wix-cli/supported-extensions/backend-extensions/api/add-api-extensions-with-the-cli).
- **Dashboard modal:** Extend your Wix app's functionality by adding modals to your app's dashboard. Learn more about [Dashboard modals](https://dev.wix.com/docs/build-apps/develop-your-app/frameworks/wix-cli/supported-extensions/dashboard-extensions/dashboard-modals/add-dashboard-modal-extensions-with-the-cli).

> - **Dashboard menu plugin:** Extend your Wix app's functionality by adding menu plugins to the user interfaces of an apps built by Wix. Learn more about [Dashboard menu plugin extensions](https://dev.wix.com/docs/build-apps/develop-your-app/extensions/dashboard-extensions/dashboard-plugins/dashboard-menu-plugins/dashboard-menu-plugin-extensions).

Reviewer: ```suggestion
- **Dashboard menu plugin:** Extend your Wix app's functionality by adding menu plugins to the user interface of an app built by Wix. Learn more about [Dashboard menu plugin extensions](https://dev.wix.com/docs/build-apps/develop-your-app/extensions/dashboard-extensions/dashboard-plugins/dashboard-menu-plugins/dashboard-menu-plugin-extensions).
```

> - **WhatsApp integration:** Prepare personalized WhatsApp messages to customers from within your app.

Reviewer: Prepare -> Send?

## Prerequisites

Before getting started, make sure you have the following set up:

+ [Node.js](https://nodejs.org/en/) (v18.16.0 or higher)
+ [A Wix developer account](https://users.wix.com/signin?loginDialogContext=signup&referralInfo=HEADER&postLogin=https:%2F%2Fdev.wix.com%2Fdc3%2Fmy-apps&postSignUp=https:%2F%2Fdev.wix.com%2Fdc3%2Fmy-apps&forceRender=true)

> + A WhatsApp account.

Reviewer: ```suggestion
+ A WhatsApp account
```

## Local Development

We first need to create local files for our app project and set up a local development environment for preview and testing.

## Step 1 | Create a new app project with the Post-Order Messaging template

Run the following command to create a new app project using this template:

> ```tsx

Reviewer: ```suggestion
```bash
```

> [fill this in]

Reviewer: don't forget to add this

```

In the creation process, you will be asked for:

> + A **Wix app name**. This is the name that appears for your app in the Wix Dev Center.

Reviewer: ```suggestion
+ A **Wix app name**. This is the name that appears for your app in the App Dashboard.
```
It's no longer the Dev Center. Please fix all references to it.

  [screenshot]()

## Extend and customize the app 

The template is designed for easy customization and extension. Here are some suggested entry points where you can add your own custom logic or functionality:

### Customize the dashboard menu plugin

You can customize the name and icon of the dashboard menu plugin, and what happens when it's clicked.

Development entry point: [`template/src/dashboard/menu-plugins/my-plugin/plugin.json`](./template/src/dashboard/menu-plugins/my-plugin/plugin.json)

You can also replace this plugin with a plugin for another of the [Order page's plugin slots](https://dev.wix.com/docs/build-apps/develop-your-app/extensions/dashboard-extensions/dashboard-plugins/supported-wix-app-dashboard-pages/wix-e-commerce/wix-e-commerce-order-page#slot-4-more-actions-menu).

### Customize the dashboard modal

Customize the dashboard modal to fit your specific needs, whether it's updating the UI or changing the functionality.

Development entry point: [`template/src/dashboard/modals/sell-message-modal/modal.tsx`](./template/src/dashboard/modals/sell-message-modal/modal.tsx)

> ### Customize the WhatsApp integration

Reviewer: Maybe the heading should be "Customize the messaging integration", since the idea is that a dev can implement here any integration with any service they like?
